pkg/cmd/cli/cmd/login.go

```diff
@@ -79,6 +85,13 @@ func NewCmdLogin(fullName string, f *osclientcmd.Factory, reader io.Reader, out
 				os.Exit(1)
 
 			} else {
+				if os.IsPermission(err) {
+					fmt.Fprintf(out, "The kubeconfig file %q is not writable. Specify a writeable path with --config or by setting $KUBECONFIG.\n", options.PathOptions.GetDefaultFilename())
+					if Username == "system:admin" {
+						fmt.Fprintln(out, "Warning: Please do not use system configuration file. This can lead to serious errors.")
+					}
+					os.Exit(1)
```

Comment:
Given the length and specificity of the error, I don't feel bad about
special casing this.  Errors are supposed to be lower case, one line and we
clearly want full sentences and newlines for this.

On Thu, Apr 7, 2016 at 10:12 AM, Steve Kuznetsov <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/login.go
> <https://github.com/openshift/origin/pull/8384#discussion_r58877600>:
>
> > @@ -79,6 +85,13 @@ func NewCmdLogin(fullName string, f *osclientcmd.Factory, reader io.Reader, out
> >  				os.Exit(1)
> >
> >  			} else {
> > +				if os.IsPermission(err) {
> > +					fmt.Fprintf(out, "The kubeconfig file %q is not writable. Specify a writeable path with --config or by setting $KUBECONFIG.\n", options.PathOptions.GetDefaultFilename())
> > +					if Username == "system:admin" {
> > +						fmt.Fprintln(out, "Warning: Please do not use system configuration file. This can lead to serious errors.")
> > +					}
> > +					os.Exit(1)
>
> We still shouldn't be doing this here -- allow the kcmdutil.CheckErr to
> handle exits.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8384/files/1469e10584d8b1c6e91a2f84663a925c57dd1b04#r58877600>
>